addons/udes_stock_refactoring/tests/test_splitting.py

```diff
@@ -0,0 +1,659 @@
+        """Reserve self.picking with two pallet of the same product and check it
+        splits correctly when reserved."""
+        MoveLine = self.env["stock.move.line"]
+
+        cherry_pallet1 = self.create_package()
+        self.create_quant(
+            self.cherry.id, self.test_stock_location_01.id, 10, package_id=cherry_pallet1.id
+        )
+        cherry_pallet2 = self.create_package()
+        self.create_quant(
+            self.cherry.id, self.test_stock_location_01.id, 10, package_id=cherry_pallet2.id
+        )
+
+        self.create_move(self.cherry, 20, self.picking)
+        self.picking.action_assign()
+
+        cherry_pallet_move_lines = MoveLine.search(
+            [("product_id.name", "=", "Test product Cherry")]
+        )
+
+        cherry_pallet1_move_line = False
+        cherry_pallet2_move_line = False
+
+        for move_line in cherry_pallet_move_lines:
+            if move_line.package_id.name == cherry_pallet1.name:
+                cherry_pallet1_move_line = move_line
+            elif move_line.package_id.name == cherry_pallet2.name:
+                cherry_pallet2_move_line = move_line
+
+        self.assertTrue(cherry_pallet1_move_line)
+        self.assertTrue(cherry_pallet2_move_line)
+
+        pick1 = cherry_pallet1_move_line.picking_id
+        pick2 = cherry_pallet2_move_line.picking_id
+
+        self.assertEqual(pick1.state, "assigned")
+        self.assertEqual(pick2.state, "assigned")
+        self.assertEqual(pick1.group_id.name, cherry_pallet1.name)
+        self.assertEqual(pick2.group_id.name, cherry_pallet2.name)
+
+        # Check we haven't mangled the moves or move_lines
+        pick1_move = pick1.move_lines
+        self.assertEqual(len(pick1_move), 1)
+        self.assertEqual(pick1_move.product_uom_qty, 10)
+
+        pick1_move_line = pick1.move_line_ids
+        self.assertEqual(len(pick1_move_line), 1)
+        self.assertEqual(pick1_move_line.product_qty, 10)
+        self.assertEqual(pick1_move_line.package_id, cherry_pallet1)
+        self.assertEqual(pick1_move_line.result_package_id, cherry_pallet1)
+        self.assertEqual(pick1_move_line.product_id, self.cherry)
+
+        pick2_move = pick2.move_lines
+        self.assertEqual(len(pick2_move), 1)
+        self.assertEqual(pick2_move.product_uom_qty, 10)
+
+        pick2_move_line = pick2.move_line_ids
+        self.assertEqual(len(pick2_move_line), 1)
+        self.assertEqual(pick2_move_line.product_qty, 10)
+        self.assertEqual(pick2_move_line.package_id, cherry_pallet2)
+        self.assertEqual(pick2_move_line.result_package_id, cherry_pallet2)
+        self.assertEqual(pick2_move_line.product_id, self.cherry)
+
+        # Check that the original picking has been reused
+        self.assertTrue(self.picking.exists())
+
+    def test_split_pallet_different_products(self):
+        """Reserve self.picking with a pallet containing two different products
+        and check it splits correctly when reserved."""
```

Comment:
Nit: Other doc strings you ended with """ on a new line :)

Reply:
Have now updated the docstrings.